modules/service/src/main/scala/gem/Service.scala

```diff
@@ -8,46 +8,45 @@ import gem.enum._
 import scalaz._, Scalaz._
 import scalaz.concurrent.Task
 
-case class Service[M[_]](xa: Transactor[M], log: Log[M], user: User[ProgramRole]) {
+final class Service[M[_]] private (private val xa: Transactor[M], val log: Log[M], val user: User[ProgramRole]) {
 
   /**
-   * Return a list of programs whose name or id contains the given substring (case-insensitive), up
-   * to a provided maximum length. The use case is an "Open" dialog.
+   * Construct a program that yields a list of `Program` whose name or id contains the given
+   * substring (case-insensitive), up to a provided maximum length.
    */
   def queryProgramsByName(substr: String, max: Int): M[List[Program[Nothing]]] =
     log.log(user, s"""queryProgramsByName("$substr", $max)""") {
       ProgramDao.selectBySubstring(s"%$substr%", max).transact(xa)
     }
 
   /**
-   * Return a program and shallow observations. The use case is an overview page showing the
-   * program, allowing the user to navigate to a selected observation.
+   * Construct a program that attempts to change the user's password, yielding `true` on success.
    */
-  def getProgram(id: Program.Id): M[Program[Observation[Nothing]]] =
-    log.log(user, s"getProgram($id)") {
-      ???
-    }
-
-  /**
-   * Fetch an observation with full sequence information. The use case is initial population of a
-   * sequence editor.
-   */
-  def getObservation(id: Observation.Id): M[Observation[Step[_ <: Instrument]]] =
-    log.log(user, s"getObservation($id)") {
-      ???
+  def changePassword(oldPassword: String, newPassword: String): M[Boolean] =
+    log.log(user, "changePassword(***, ***)") {
+      UserDao.changePassword(user.id, oldPassword, newPassword).transact(xa)
     }
 
 }
 
 object Service {
 
-  def forTesting(uname: String): Service[Task] = {
-    val xa = DriverManagerTransactor[Task]("org.postgresql.Driver","jdbc:postgresql:gem","postgres","")
-    val io = for {
-      user <- UserDao.selectWithRoles(uname).transact(xa)
-      log  <- Log.newLog[Task]("Testing", xa)
-    } yield Service(xa, log, user)
-    io.unsafePerformSync
+  object L {
+    def user[M[_]]: Service[M] @> User[ProgramRole] = Lens.lensu((a, b) => new Service(a.xa, a.log, b), _.user)
   }
 
+  def apply[M[_]](xa: Transactor[M], log: Log[M], user: User[ProgramRole]): Service[M] =
+    new Service(xa, log, user)
+
+  /**
+   * Construct a program that verifies a user's id and password and returns a `Service`.
+   */
+  def tryLogin[M[_]: Monad: Catchable: Capture](
```

Comment:
Could you expand a bit on `Catchable` and `Capture` they seem to be defined in `scalaz`